tectonic/xetex-ini.c

```diff
@@ -1198,6 +1220,9 @@ new_hyph_exceptions(void)
                     hc[0] = trie_tro[hyph_index + cur_chr];
 
                 if (hc[0] == 0) {
+                    ttstub_diag_finish(diagnostic_error_here(
+                        "Not a letter"));
+
```

Comment:
I'm seeing a lot of cases where you're duplicating the message text (i.e. "Not a letter"). I'd prefer to avoid that if possible. From a quick browser, it looks like it might be helpful to add a helper function of the form:

```
{
  ttstub_diag_finish(diagnostic_error_here(message));
  if (file_line_error_style_p)
    print_file_line();
  else
    print_nl_cstr("! ");
  print_cstr(message);
}
```

Reply:
I've added `error_here_with_diagnostic` in xetex-output.c that creates a diagnostic, prints the line number, and sends a message to both. I'm not sure about the name though.